Content.Shared/MedicalScanner/HealthAnalyzerUiState.cs

```diff
@@ -0,0 +1,50 @@
+﻿using Content.Shared.Damage;
+using Content.Shared.FixedPoint;
+using Content.Shared.IdentityManagement;
+using Robust.Shared.Serialization;
+
+namespace Content.Shared.MedicalScanner;
+
+[Serializable, NetSerializable]
+public sealed class HealthAnalyzerUiState : BoundUserInterfaceState
+{
+    public string EntityName = "Unknown";
+
+    public float Temperature;
+    public float BloodLevel;
+
+    public DamageDetails? Damage;
+
+    public HealthAnalyzerUiState(HealthAnalyzerScannedUserMessage msg)
+    {
+        var entities = IoCManager.Resolve<IEntityManager>();
+
+        Temperature = msg.Temperature;
+        BloodLevel = msg.BloodLevel;
+
+        if (msg.TargetEntity != null &&
+            entities.TryGetComponent<MetaDataComponent>(msg.TargetEntity.Value, out var metaData))
+            EntityName = Identity.Name(msg.TargetEntity.Value, entities);
+
+        if (msg.TargetEntity == null ||
+            !entities.TryGetComponent<DamageableComponent>(msg.TargetEntity, out var damageable))
+            return;
```

Comment:
You shouldn't ioc resolve and call `TryGetComponent` inside the constructor for ui states. Move the try gets to where `HealthAnalyzerUiState` is instantiated. You can modify the `HealthAnalyzerSystem` to send this state instead of whatever it sends now if you need to.